### ISSUE TITLE
The RemoveExplicitCallToSuper mutator removes important this(); calls

### ISSUE THREAD

**Issue by shrugalic** [solven-eu/cleanthat#804]
# Setup
We just added `cleanthat` to our Gradle build of a Java 17 project, via [`spotless`](https://github.com/diffplug/spotless/tree/main/plugin-gradle#cleanthat) (v6.25.0), including the `SafeButNotConsensual ` mutator:
```kts
cleanthat()
    .version("2.18")
    .sourceCompatibility("17")
    .addMutator("SafeButNotConsensual")
```

# Issue / How to reproduce
The `RemoveExplicitCallToSuper` mutator produces erroneous code that does not compile. Here's an example:
```java
public class Sample {

    private String stringField;

    private final int intField;

    public Sample() {
        stringField = "default";
        intField = 42;
    }

    public Sample(String otherValue) {
        this(); // This will be removed by RemoveExplicitCallToSuper and not compile afterwards
        stringField = otherValue;
    }
}
```

# Expecation
The `this();` call is important and must remain.

# Issue Localization
When adding `.excludeMutator("RemoveExplicitCallToSuper")` to the configuration, it does not happen.

# Superficial look
I'm not at all familiar with this code base, and haven't looked into the details, but from a superficial look it seems [removing just any old firstStatement in a constructor](https://github.com/solven-eu/cleanthat/blob/master/java/src/main/java/eu/solven/cleanthat/engine/java/refactorer/mutators/RemoveExplicitCallToSuper.java#L97) seems a bit optimistic?

**Comment by blacelle** [comment]
Thanks for the report.

> but from a superficial look it seems [removing just any old firstStatement in a constructor](https://github.com/solven-eu/cleanthat/blob/master/java/src/main/java/eu/solven/cleanthat/engine/java/refactorer/mutators/RemoveExplicitCallToSuper.java#L97) seems a bit optimistic?

I confirm this statement, and this issue. It has been detected some time ago, and it seems it is not fixed yet.

**Comment by blacelle** [comment]
This is fixed in cleanthat 2.19. I dropped constructor reference to `this()` as it looks like a plain misbehavior.

**Comment by shrugalic** [comment]
I successfully verified that the issue was fixed, thanks a lot. 🙏
